misc/get-doc.js

```diff
@@ -1,6 +1,38 @@
-var doc = require('./api-doc');
+const fs = require('fs');
+const doc = require('./api-doc');
+const excluded = new Set([
+  'utils'
+]);
+
+function isDirectory(path) {
+  return fs.lstatSync(path).isDirectory();
+}
 
 function getFileNames(directiveName) {
   return [`./src/${directiveName}/${directiveName}.ts`];
 }
-console.log(JSON.stringify(doc(getFileNames('tabset'))));
+
+function fileExists(file) {
+  let result = false;
+  try {
+    fs.accessSync(file);
+    result = true;
+  }
+  catch (e) {
+    console.error(`the specified file does not exist: ${file}`)
+  }
+
+  return result;
+}
+
+const componentMap = new Map();
+const items = fs.readdirSync('./src');
+items.forEach(component => {
+  if (!excluded.has(component) && isDirectory(`./src/${component}`)) {
+    const names = getFileNames(component);
+    if (names.every(name => fileExists(name))) {
+      componentMap.set(component, JSON.stringify(doc(names)));
```

Comment:
Why are you calling `JSON.stringify` here? Isn't it going to make your life harder when generating API docs?